integration_tests/version_generation_test.py

```diff
@@ -98,7 +98,29 @@ def extracted_sdist(
 
         with tempfile.TemporaryDirectory() as temp_dir:
             with tarfile.open(str(sdist_tarbal_path), 'r') as sdist_tarbal:
-                sdist_tarbal.extractall(temp_dir)
+                
+                import os
+                
+                def is_within_directory(directory, target):
+                    
+                    abs_directory = os.path.abspath(directory)
+                    abs_target = os.path.abspath(target)
+                
+                    prefix = os.path.commonprefix([abs_directory, abs_target])
+                    
+                    return prefix == abs_directory
+                
+                def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+                
+                    for member in tar.getmembers():
+                        member_path = os.path.join(path, member.name)
+                        if not is_within_directory(path, member_path):
+                            raise Exception("Attempted Path Traversal in Tar File")
```

Comment:
Do not raise Exception but tarfile.TarError (or a derived error): https://docs.python.org/3/library/tarfile.html#tarfile.TarError

Reply:
Thanks for the PR.

The code using tarfile.extractall is test code that only receives tarfiles created by `python setup.py` / `python -m build` during the tests. You are correct that the code does not verify that the extracted tar files are safe but as all of those are created in the tests this isn't that big of on issue.

Please run tests and static analyzers using tox (sorry that there is no CI (yet)).

Pylint reports the following issues:
```
************* Module integration_tests.version_generation_test
integration_tests/version_generation_test.py:101:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:103:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:105:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:108:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:110:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:112:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:114:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:119:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:120:78: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:121:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:122:0: C0303: Trailing whitespace (trailing-whitespace)
integration_tests/version_generation_test.py:102:16: C0415: Import outside toplevel (os) (import-outside-toplevel)
```